databricks/koalas/namespace.py

```diff
@@ -672,6 +675,26 @@ def read_parquet(path, columns=None, index_col=None, **options) -> DataFrame:
     if columns is not None:
         columns = list(columns)
 
+    index_names = None
+
+    if index_col is None and LooseVersion(pyspark.__version__) >= LooseVersion("3.0.0"):
+        # Try to read pandas metadata
+        binary = default_session().read.format("binaryFile").load(path).select("content").head()[0]
```

Comment:
My only concern is if it's worthwhile enough to have the overhead.  It adds some extra overhead when `read_parquet` is from Spark or Koalas (not from pandas).

If people read it from the file written in pandas more often, it makes sense.
If people read it from Spark / Koalas more often, I think it might not be worthwhile.

If the file is big, fetching the entire file into the driver side won't be a trivial overhead.

Reply:
Good point.
Maybe shall we have a config?